.github/workflows/int_test.yml

```diff
@@ -36,38 +36,62 @@ on:
         required: false
         default: false
 
-env:
-  CLUSTER_NAME: "az-iot-ops-test-cluster-${{ github.run_id }}-${{ github.run_attempt }}"
-  KV_NAME: "opstestkv-${{ github.run_id }}-${{ github.run_attempt }}"
-  CLUSTER_RG: ${{ inputs.resource_group }}
-  CUSTOM_LOCATIONS_OID: "51dfe1e8-70c6-4de5-a08e-e18aff23d815"
-  EXTENSION_SOURCE_DIRECTORY: "./azure-iot-ops-cli-extension"
-  K3S_VERSION: "v1.28.5+k3s1"
-
 permissions:
   # required for OpenID federation
   contents: 'read'
   id-token: 'write'
 
+env:
+  KV_NAME: "opskv${{ github.run_number }}"
+  RESOURCE_GROUP: "${{ inputs.resource_group }}"
+
 jobs:
-  integration_test:
-    name: Integration test
+  create_kv:
     runs-on: ubuntu-22.04
-    outputs:
-      cluster_name: ${{ env.CLUSTER_NAME }}
-      resource_group: ${{ env.CLUSTER_RG }}
-      keyvault_name: ${{ env.KV_NAME }}
+    steps:
+    - name: "Az CLI login"
+      uses: azure/login@v1
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+    - name: "Create Key Vault for clusters"
+      run: az keyvault create -n ${{ env.KV_NAME }} -g ${{ env.RESOURCE_GROUP }} --enable-rbac-authorization false
+    
+  test:
+    needs: [create_kv]
+    env:
+      CLUSTER_NAME: "az-iot-ops-test-cluster${{ github.run_number }}${{ matrix.feature }}"
+      CUSTOM_LOCATIONS_OID: "51dfe1e8-70c6-4de5-a08e-e18aff23d815"
```

Comment:
Should this be an Action variable since its dynamic based on the tenant?

Reply:
It should be, yeah - in the actions workflow (action.yml) it is currently an optional input, and if not provided we try to get it using your az login credentials.

But since this workflow can be called externally, probably a good idea to promote this as a parameter as well